javascript/ql/lib/semmle/javascript/JSON.qll

```diff
@@ -54,7 +54,9 @@ class JsonValue extends @json_value, Locatable {
   int getIntValue() { result = this.(JsonNumber).getValue().toInt() }
 
   /** If this is a boolean constant, gets its boolean value. */
-  boolean getBooleanValue() { result.toString() = this.(JsonBoolean).getValue() }
+  boolean getBooleanValue() {
+    result.toString() = this.(JsonBoolean).getValue() and result = [true, false]
```

Comment:
It's actually not implied by the return type, because `boolean` is a primitive, and CodeQL treats all primitives as unbounded.   
(That makes more sense for `string` and `number`, but it's also the case for `boolean`).  

In other places there is a `class Boolean extends boolean { Boolean() { this = [true, false]}}` to fix the same issue.

Reply:
I'm glad I asked :)